server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -254,6 +247,17 @@ public final void addDynamicMapper(Mapper mapper) {
         dynamicMappers.add(mapper);
     }
 
+    private void checkFieldLimit(String fieldName) {
+        // eagerly check field name limit here to avoid OOM errors
+        // only check fields that are not already mapped or tracked in order to avoid hitting field limit too early via double-counting
+        // note that existing fields can also receive dynamic mapping updates (e.g. constant_keyword to fix the value)
+        if (mappingLookup.getMapper(fieldName) == null
+            && mappingLookup.objectMappers().containsKey(fieldName) == false
+            && newFieldsSeen.add(fieldName)) {
```

Comment:
Right, based on the comments, since existing mappers can receive mapping updates, we needed this check.

> When we are adding a dynamic runtime field these these two checks are redundant, because they will always return false? Do we want to have two different codepaths then for the two scenarios?

Thanks for explaining this, I did not know this. I've made the tow different codepaths in 1ee42705cb6ad4c4c5b6083a33c042a1b8269912